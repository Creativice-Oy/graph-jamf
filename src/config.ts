import {
  IntegrationInstanceConfig,
  IntegrationInstanceConfigFieldMap,
} from '@jupiterone/integration-sdk-core';

export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
  jamfHost: {
    type: 'string',
  },
  jamfUsername: {
    type: 'string',
  },
  jamfPassword: {
    type: 'string',
    mask: true,
  },
};

export interface IntegrationConfig extends IntegrationInstanceConfig {
  jamfHost: string;
  jamfUsername: string;
  jamfPassword: string;
}
