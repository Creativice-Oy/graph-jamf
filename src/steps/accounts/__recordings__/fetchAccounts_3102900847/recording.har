{
  "log": {
    "_recordingName": "fetchAccounts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "4f0e505db41b492155ff6d75affa0642",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "identity"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiteronedev.jamfcloud.com"
            }
          ],
          "headersSize": 364,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://jupiteronedev.jamfcloud.com/api/v1/auth/token"
        },
        "response": {
          "bodySize": 392,
          "content": {
            "mimeType": "application/json",
            "size": 392,
            "text": "{\"token\":\"[REDACTED]\",\"expires\":\"2023-01-18T20:30:08.913Z\"}"
          },
          "cookies": [
            {
              "expires": "2023-01-25T20:00:08.000Z",
              "name": "AWSALB",
              "path": "/",
              "value": "[REDACTED]"
            },
            {
              "expires": "2023-01-25T20:00:08.000Z",
              "name": "AWSALBCORS",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "expires": "1970-01-01T00:00:00.000Z",
              "httpOnly": true,
              "maxAge": 0,
              "name": "JSESSIONID",
              "path": "/",
              "sameSite": "Lax",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "APBALANCEID",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 18 Jan 2023 20:00:08 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "Jamf Cloud Node"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains;, max-age=31536000 ; includeSubDomains"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            }
          ],
          "headersSize": 969,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-01-18T20:00:08.345Z",
        "time": 670,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 670
        }
      },
      {
        "_id": "e0cc2e06750b002c84a3dde729e7f689",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "identity"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiteronedev.jamfcloud.com"
            }
          ],
          "headersSize": 639,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://jupiteronedev.jamfcloud.com/api/v1/jamf-pro-version"
        },
        "response": {
          "bodySize": 39,
          "content": {
            "mimeType": "application/json",
            "size": 39,
            "text": "{\"version\":\"10.42.1-t1667311080\"}"
          },
          "cookies": [
            {
              "expires": "2023-01-25T20:00:09.000Z",
              "name": "AWSALB",
              "path": "/",
              "value": "[REDACTED]"
            },
            {
              "expires": "2023-01-25T20:00:09.000Z",
              "name": "AWSALBCORS",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "APBALANCEID",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 18 Jan 2023 20:00:09 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "Jamf Cloud Node"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains;, max-age=31536000 ; includeSubDomains"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            }
          ],
          "headersSize": 855,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-01-18T20:00:09.024Z",
        "time": 434,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 434
        }
      },
      {
        "_id": "25958b0444ec2a193681bde268b4670d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "identity"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiteronedev.jamfcloud.com"
            }
          ],
          "headersSize": 636,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://jupiteronedev.jamfcloud.com/JSSResource/accounts"
        },
        "response": {
          "bodySize": 692,
          "content": {
            "mimeType": "text/plain;charset=UTF-8",
            "size": 692,
            "text": "{\"accounts\":{\"users\":[{\"id\":12,\"name\":\"adam.pierson\"},{\"id\":1,\"name\":\"adam.williams\"},{\"id\":9,\"name\":\"austin.kelleher\"},{\"id\":13,\"name\":\"chasen.bettinger\"},{\"id\":3,\"name\":\"erkang.zheng\"},{\"id\":7,\"name\":\"isaacwilliams\"},{\"id\":8,\"name\":\"jupiterone.dev\"},{\"id\":14,\"name\":\"max.daum\"},{\"id\":17,\"name\":\"my.new.account\"},{\"id\":10,\"name\":\"nick.dowmon\"},{\"id\":15,\"name\":\"samuel.poulton\"},{\"id\":11,\"name\":\"stefan.virag\"},{\"id\":16,\"name\":\"Wrong_Permissions\"}],\"groups\":[{\"id\":1,\"name\":\"Test Group\",\"site\":{\"id\":1,\"name\":\"Test site\"}},{\"id\":2,\"name\":\"Another Test Group\",\"site\":{\"id\":1,\"name\":\"Test site\"}},{\"id\":3,\"name\":\"Not all groups have members inside in api\",\"site\":{\"id\":1,\"name\":\"Test site\"}}]}}"
          },
          "cookies": [
            {
              "expires": "2023-01-25T20:00:09.000Z",
              "name": "AWSALB",
              "path": "/",
              "value": "[REDACTED]"
            },
            {
              "expires": "2023-01-25T20:00:09.000Z",
              "name": "AWSALBCORS",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "APBALANCEID",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 18 Jan 2023 20:00:09 GMT"
            },
            {
              "name": "content-type",
              "value": "text/plain;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "692"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "Jamf Cloud Node"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains;, max-age=31536000 ; includeSubDomains"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "cache-control",
              "value": "no-store, no-cache, must-revalidate, max-age=0, post-check=0, pre-check=0"
            },
            {
              "name": "accept-ranges",
              "value": "bytes"
            },
            {
              "name": "vary",
              "value": "Accept-Charset,Accept-Encoding,Accept-Language,Accept"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            }
          ],
          "headersSize": 964,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-01-18T20:00:09.498Z",
        "time": 477,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 477
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
