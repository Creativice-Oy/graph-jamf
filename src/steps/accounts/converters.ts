import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Account } from '../../jamf/types';
import { generateEntityKey } from '../../util/generateKey';
import { Entities } from '../constants';

export function getAccountEntityKey(accountId: string) {
  return generateEntityKey(Entities.ACCOUNT._type, accountId);
}

export function createAccountEntity(data: Account) {
  return createIntegrationEntity({
    entityData: {
      source: data,
      assign: {
        _class: Entities.ACCOUNT._class,
        _type: Entities.ACCOUNT._type,
        _key: getAccountEntityKey(data.id),
        displayName: data.name,
        name: data.name,
      },
    },
  });
}
