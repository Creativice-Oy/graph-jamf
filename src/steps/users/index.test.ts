import { fetchAdminUsers, fetchDeviceUsers } from '.';
import { createDataCollectionTest } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { Entities, Relationships } from '../constants';
import { fetchAccounts } from '../accounts';
import { RelationshipClass } from '@jupiterone/integration-sdk-core';
import {
  setComputerDeviceIdToGraphObjectKeyMap,
  setMobileDeviceIdToGraphObjectKeyMap,
} from '../../util/device';

describe('#fetchAdminUsers', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchAdminUsers',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchAccounts, fetchAdminUsers],
      entitySchemaMatchers: [
        {
          _type: Entities.USER_ADMIN._type,
          matcher: {
            _class: ['User'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'jamf_user' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                username: { type: 'string' },
                admin: {
                  const: true,
                },
                directoryUser: { type: 'boolean' },
                fullName: { type: 'string' },
                email: { type: 'string' },
                emailAddress: { type: 'string' },
                enabled: { type: 'string' },
                forcePasswordChange: { type: 'boolean' },
                accessLevel: { type: 'string' },
                privilegeSet: { type: 'string' },
                permissions: {
                  type: 'array',
                  items: { type: 'string' },
                },
              },
            },
          },
        },
      ],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.ACCOUNT_HAS_USER_ADMIN._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.HAS },
                _type: { const: 'jamf_account_has_user' },
              },
            },
          },
        },
      ],
    });
  }, 10000);
});

describe('#fetchDeviceUsers', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      async beforeRecord(context) {
        await setMobileDeviceIdToGraphObjectKeyMap(context.jobState, new Map());
        await setComputerDeviceIdToGraphObjectKeyMap(
          context.jobState,
          new Map(),
        );
      },
      recordingName: 'fetchDeviceUsers',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchAccounts, fetchDeviceUsers],
      entitySchemaMatchers: [
        {
          _type: Entities.DEVICE_USER._type,
          matcher: {
            _class: ['User'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'device_user' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                username: { type: 'string' },
                fullName: { type: 'string' },
                email: { type: 'string' },
                emailAddress: { type: 'string' },
                phoneNumber: { type: 'string' },
                position: { type: 'string' },
                enableCustomPhotoUrl: { type: 'boolean' },
                customPhotoUrl: { type: 'string' },
                ldapServer: { type: 'string' },
                totalVppCodeCount: { type: 'number' },
                os: {
                  type: 'array',
                  items: { type: 'string' },
                },
              },
            },
          },
        },
      ],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.ACCOUNT_HAS_DEVICE_USER._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.HAS },
                _type: { const: 'jamf_account_has_device_user' },
              },
            },
          },
        },
      ],
    });
  }, 10000);
});
