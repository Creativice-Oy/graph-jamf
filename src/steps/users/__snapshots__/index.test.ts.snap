// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAdminUsers should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "jamf_account_local-integration-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "local-integration-instance",
            "name": "Local Integration",
          },
        },
      ],
      "_type": "jamf_account",
      "createdOn": undefined,
      "displayName": "Local Integration",
      "id": "local-integration-instance",
      "name": "Local Integration",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "jamf_user_1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "jamf_user",
      "accessLevel": "Full Access",
      "admin": true,
      "createdOn": undefined,
      "directoryUser": false,
      "displayName": "adam.williams",
      "email": "adam.williams@jupiterone.com",
      "emailAddress": "adam.williams@jupiterone.com",
      "enabled": "Enabled",
      "forcePasswordChange": false,
      "fullName": "",
      "id": "1",
      "name": "adam.williams",
      "permissions": Array [
        "Administrator",
      ],
      "privilegeSet": "Administrator",
      "username": "adam.williams",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "jamf_user_9",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "jamf_user",
      "accessLevel": "Full Access",
      "admin": true,
      "createdOn": undefined,
      "directoryUser": false,
      "displayName": "austin.kelleher",
      "email": "austin.kelleher@jupiterone.com",
      "emailAddress": "austin.kelleher@jupiterone.com",
      "enabled": "Enabled",
      "forcePasswordChange": false,
      "fullName": "Austin Kelleher",
      "id": "9",
      "name": "austin.kelleher",
      "permissions": Array [
        "Administrator",
      ],
      "privilegeSet": "Administrator",
      "username": "austin.kelleher",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "jamf_user_3",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "jamf_user",
      "accessLevel": "Full Access",
      "admin": true,
      "createdOn": undefined,
      "directoryUser": false,
      "displayName": "erkang.zheng",
      "email": "erkang.zheng@lifeomic.com",
      "emailAddress": "erkang.zheng@lifeomic.com",
      "enabled": "Enabled",
      "forcePasswordChange": false,
      "fullName": "Erkang Zheng",
      "id": "3",
      "name": "erkang.zheng",
      "permissions": Array [
        "Administrator",
      ],
      "privilegeSet": "Administrator",
      "username": "erkang.zheng",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "jamf_user_7",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "jamf_user",
      "accessLevel": "Full Access",
      "admin": true,
      "createdOn": undefined,
      "directoryUser": false,
      "displayName": "isaacwilliams",
      "email": "isaac.williams@lifeomic.com",
      "emailAddress": "isaac.williams@lifeomic.com",
      "enabled": "Enabled",
      "forcePasswordChange": false,
      "fullName": "Isaac Williams",
      "id": "7",
      "name": "isaacwilliams",
      "permissions": Array [
        "Administrator",
      ],
      "privilegeSet": "Administrator",
      "username": "isaacwilliams",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "jamf_user_8",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "jamf_user",
      "accessLevel": "Full Access",
      "admin": true,
      "createdOn": undefined,
      "directoryUser": false,
      "displayName": "jupiterone.dev",
      "email": undefined,
      "emailAddress": "",
      "enabled": "Enabled",
      "forcePasswordChange": false,
      "fullName": "JupiterOne API Development",
      "id": "8",
      "name": "jupiterone.dev",
      "permissions": undefined,
      "privilegeSet": "Custom",
      "username": "jupiterone.dev",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|jamf_user_1",
      "_toEntityKey": "jamf_user_1",
      "_type": "jamf_account_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|jamf_user_9",
      "_toEntityKey": "jamf_user_9",
      "_type": "jamf_account_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|jamf_user_3",
      "_toEntityKey": "jamf_user_3",
      "_type": "jamf_account_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|jamf_user_7",
      "_toEntityKey": "jamf_user_7",
      "_type": "jamf_account_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|jamf_user_8",
      "_toEntityKey": "jamf_user_8",
      "_type": "jamf_account_has_user",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "jamf_account",
    "jamf_user",
    "jamf_account_has_user",
  ],
  "numCollectedEntities": 6,
  "numCollectedRelationships": 5,
}
`;

exports[`#fetchDeviceUsers should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "jamf_account_local-integration-instance",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "id": "local-integration-instance",
            "name": "Local Integration",
          },
        },
      ],
      "_type": "jamf_account",
      "createdOn": undefined,
      "displayName": "Local Integration",
      "id": "local-integration-instance",
      "name": "Local Integration",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_5",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "Heriberto Truby",
      "email": "heriberto.truby@gmail.com",
      "emailAddress": "heriberto.truby@gmail.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "",
      "id": "5",
      "ldapServer": "None",
      "name": "Heriberto Truby",
      "os": Array [],
      "phoneNumber": "",
      "position": "",
      "totalVppCodeCount": 0,
      "username": "Heriberto Truby",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_2",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "Lael Buresh",
      "email": "lael.buresh@gmail.com",
      "emailAddress": "lael.buresh@gmail.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "",
      "id": "2",
      "ldapServer": "None",
      "name": "Lael Buresh",
      "os": Array [],
      "phoneNumber": "",
      "position": "",
      "totalVppCodeCount": 0,
      "username": "Lael Buresh",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_4",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "Maira Fillman",
      "email": "maira.fillman@gmail.com",
      "emailAddress": "maira.fillman@gmail.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "",
      "id": "4",
      "ldapServer": "None",
      "name": "Maira Fillman",
      "os": Array [],
      "phoneNumber": "",
      "position": "",
      "totalVppCodeCount": 0,
      "username": "Maira Fillman",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_3",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "Rory Overbey",
      "email": "rory.overbey@gmail.com",
      "emailAddress": "rory.overbey@gmail.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "",
      "id": "3",
      "ldapServer": "None",
      "name": "Rory Overbey",
      "os": Array [],
      "phoneNumber": "",
      "position": "",
      "totalVppCodeCount": 0,
      "username": "Rory Overbey",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "Test",
      "email": "test@gmail.com",
      "emailAddress": "test@gmail.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "Test User",
      "id": "1",
      "ldapServer": "None",
      "name": "Test",
      "os": Array [],
      "phoneNumber": "",
      "position": "",
      "totalVppCodeCount": 0,
      "username": "Test",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "device_user_6",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "uploadStatus": "SKIPPED",
            "uploadStatusReason": "Raw data currently disabled for this entity type",
          },
        },
      ],
      "_type": "device_user",
      "createdOn": undefined,
      "customPhotoUrl": undefined,
      "displayName": "erkang.zheng",
      "email": "erkang.zheng@lifeomic.com",
      "emailAddress": "erkang.zheng@lifeomic.com",
      "enableCustomPhotoUrl": undefined,
      "fullName": "Erkang Zheng",
      "id": "6",
      "ldapServer": "None",
      "name": "erkang.zheng",
      "os": Array [
        "macOS",
      ],
      "phoneNumber": "9195228915",
      "position": "CISO",
      "totalVppCodeCount": 0,
      "username": "erkang.zheng",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_5",
      "_toEntityKey": "device_user_5",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_2",
      "_toEntityKey": "device_user_2",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_4",
      "_toEntityKey": "device_user_4",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_3",
      "_toEntityKey": "device_user_3",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_1",
      "_toEntityKey": "device_user_1",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jamf_account_local-integration-instance",
      "_key": "jamf_account_local-integration-instance|has|device_user_6",
      "_toEntityKey": "device_user_6",
      "_type": "jamf_account_has_device_user",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "jamf_account",
    "device_user",
    "jamf_account_has_device_user",
  ],
  "numCollectedEntities": 7,
  "numCollectedRelationships": 6,
}
`;
